rust/tools/authorization-logic/src/parsing/astconstructionvisitor.rs

```diff
@@ -0,0 +1,243 @@
+/*
+ * Copyright 2021 The Raksha Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+use crate::{
+    ast::*,
+    parsing::antlr_gen::{authlogiclexer::*, authlogicparser::*},
+};
+use antlr_rust::{common_token_stream::CommonTokenStream, tree::ParseTree, InputStream};
+
+/// The only public interface to the parser is `parse_program`. It produces an
```

Comment:
(nit) no need to repeat the function name in the doc, it is already implicitly associated with that.

Reply:
Ah thanks, didn't know. Applied suggestion broadly (and also with files).